Experiment_Launcher_code/ExperimentManager.py

```diff
@@ -243,52 +267,57 @@ def start_streaming_exp(self, experiment_name, num_trial, decision_time, return_
             mouse1 = FixedStrategyPrisoner(opponent1_strategy, probability=0.8)
             mouse2 = FixedStrategyPrisoner(opponent2_strategy, probability=0.8)
             self.opponent_path = "COMPUTER_COMPUTER"
-        self.trial_logger.start_logging(self.mouse_number,self.opponent_path)
-        force_end_time = num_trial * (decision_time + return_time)
-        self.stateManager.SetTimeOut(decision_time, return_time )
+        self.trial_logger.start_logging(self.mouse_number, self.opponent_path)
+
+        self.stateManager.SetTimeOut(decision_time, return_time)
 
         currentstate = None
         state_history = []
-
+        listener = Listener(on_press=self.on_press, on_release=self.on_release)
+        listener.start()
         while currentstate != States.End:
-            trialevents = 0
+
+            self.trialevents = 0
```

Comment:
You will loose the user key press here.
Make trialevent a local variable as before - not self.variable. Its not a good idea to shrae variables between functions which are working _asynchronously_. It is always a mess.